api/v1alpha2/contentlibraryitem_types.go

```diff
@@ -0,0 +1,374 @@
+
+// VirtualControllerType defines the allowed types of virtual controllers.
+type VirtualControllerType string
+
+const (
+	VirtualControllerTypeParaVirtualSCSI VirtualControllerType = "ParaVirtualSCSI"
+	VirtualControllerTypeBusLogic        VirtualControllerType = "BusLogic"
+	VirtualControllerTypeLsiLogic        VirtualControllerType = "LsiLogic"
+	VirtualControllerTypeLsiLogicSAS     VirtualControllerType = "LsiLogicSAS"
+	VirtualControllerTypeTypeNVME        VirtualControllerType = "NVME"
+	VirtualControllerTypeTypeAHCI        VirtualControllerType = "AHCI"
+)
+
+type ContentLibraryItemControllerStatus struct {
+	// Type describes the type of controller.
+	Type VirtualControllerType `json:"type"`
+
+	// Key describes the controller's device key.
+	Key int32 `json:"key"`
+}
+
+// +kubebuilder:validation:Enum=FlatVer1;FlatVer2;SeSparse;SparseVer1;SparseVer2
+
+// VirtualDiskType defines the allowed types of virtual disks.
+type VirtualDiskType string
+
+const (
+	VirtualDiskTypeFlatVer1   VirtualDiskType = "FlatVer1"
+	VirtualDiskTypeFlatVer2   VirtualDiskType = "FlatVer2"
+	VirtualDiskTypeSeSparse   VirtualDiskType = "SeSparse"
+	VirtualDiskTypeSparseVer1 VirtualDiskType = "SparseVer1"
+	VirtualDiskTypeSparseVer2 VirtualDiskType = "SparseVer2"
+)
+
+type VirtualDiskInfo struct {
+	// Key describes the disk's device key.
+	Key int32 `json:"key"`
+
+	// ControllerKey describes the device key of the controller to which the
+	// disk is connected.
+	ControllerKey int32 `json:"controllerKey"`
+
+	// +optional
+
+	// Capacity describes the capacity of the disk.
+	Capacity resource.Quantity `json:"capacity,omitempty"`
+
+	// Type describes the type of virtual disk.
+	Type VirtualDiskType `json:"type"`
+}
+
+// FileInfo represents the information of a file in a content library item in
+// vCenter.
+type FileInfo struct {
+	// Name describes the name of the file.
+	Name string `json:"name"`
+
+	// +optional
+
+	// SizeInBytes describes the total number of bytes used by the file on disk.
+	SizeInBytes resource.Quantity `json:"sizeInBytes,omitempty"`
+
+	// +optional
+
+	// Version describes the version of the library item file.
+	// This value is incremented when the file is modified on disk.
+	Version string `json:"version,omitempty"`
+
+	// +optional
+
+	// Cached describes whether or not the file is available locally on disk.
+	Cached bool `json:"cached,omitempty"`
+
+	// +optional
+
+	// StorageURI describes the fully-qualified path to the file on a datastore,
+	// ex. "[my-datastore-1] library-1/item-1/file1.vmdk".
+	// This URL is useful for creating a device that is backed by this file
+	// (i.e. mounting an ISO file via a virtual CD-ROM device).
+	StorageURI string `json:"storageURI,omitempty"`
+
+	// +optional
+
+	// DiskInfo describes information about the file if it is a virtual disk,
+	// otherwise this field is omitted.
+	DiskInfo *VirtualDiskInfo `json:"diskInfo,omitempty"`
```

Comment:
comment it only applies to Inventory library type ?

Reply:
But it won't. It will also apply to type OVF since we can get it from those as well.